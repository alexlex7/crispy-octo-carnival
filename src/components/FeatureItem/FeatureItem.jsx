import React from 'react';
import styles from './FeatureItem.module.css';
import PropTypes from 'prop-types';

export default function FeatureItem({ item }) {
  const { title, text, imageURL, mobileImageURL } = item;
  return (
    <li className={styles.listItem}>
      <div className={styles.thumb}>
        <img
          srcSet={`${mobileImageURL} 28w, ${imageURL} 50w`}
          sizes="(min-width: 1205px) 50px, (max-width: 1204px) 28px, 28px)"
          src={mobileImageURL}
          alt={title}
          title={title}
        />
      </div>
      <div>
        <h6 className={styles.title}>{title}</h6>
        <p className={styles.text}>{text}</p>
      </div>
    </li>
  );
}

FeatureItem.propTypes = {
  item: PropTypes.shape({
    title: PropTypes.string.isRequired,
    text: PropTypes.string.isRequired,
    imageURL: PropTypes.string.isRequired,
    mobileImageURL: PropTypes.string.isRequired,
  }).isRequired,
};
