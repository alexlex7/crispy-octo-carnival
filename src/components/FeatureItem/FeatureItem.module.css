.listItem {
  display: flex;
  column-gap: 16px;
}

.thumb {
}

.title {
  padding: 0;
  margin: 0px 0px 4px 0px;
  font-family: 'Montserrat';
  font-style: normal;
  font-weight: 600;
  font-size: 14px;
  line-height: 20px;
  color: #244d4d;
}

.text {
  padding: 0;
  margin: 0;
  width: 151px;
  font-family: 'DM Sans';
  font-style: normal;
  font-weight: 400;
  font-size: 10px;
  line-height: 15px;
  color: #383738;
}

@media screen and (min-width: 1205px) {
  .listItem {
    column-gap: 28px;
  }
  .title {
    margin-bottom: 14px;

    font-weight: 700;
    font-size: 20px;
    line-height: 26px;
  }
  .text {
    width: 272px;

    font-size: 18px;
    line-height: 27px;
  }
}
