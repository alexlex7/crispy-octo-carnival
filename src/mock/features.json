{
  "title": "About Us",
  "description": "All of our furniture uses the best materials and choices for our customers.All of our furniture uses the best materials",
  "features": [
    {
      "title": "Best Quality",
      "text": "All of our furniture uses the best materials and choices",
      "imageURL": "http://localhost:8080/images/features/quality.png",
      "mobileImageURL": "http://localhost:8080/images/features/mobileQuality.png"
    },
    {
      "title": "Free Shipping",
      "text": "All of our furniture uses the best materials and choices",
      "imageURL": "http://localhost:8080/images/features/best.png",
      "mobileImageURL": "http://localhost:8080/images/features/mobileBest.png"
    },
    {
      "title": "Free Shipping",
      "text": "All of our furniture uses the best materials and choices",
      "imageURL": "http://localhost:8080/images/features/shipping.png",
      "mobileImageURL": "http://localhost:8080/images/features/mobileShipping.png"
    }
  ]
}
