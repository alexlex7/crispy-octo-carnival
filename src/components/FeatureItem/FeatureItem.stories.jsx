import React from 'react';
import FeatureItem from './FeatureItem';
import { features } from '../../mock/features.json';

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
  title: 'About/FeatureItem',
  component: FeatureItem,
  // More on argTypes: https://storybook.js.org/docs/react/api/argtypes
};

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
const Template = (args) => <FeatureItem {...args} />;

export const Default = Template.bind({});
// More on args: https://storybook.js.org/docs/react/writing-stories/args
Default.args = {
  item: features[0],
};
