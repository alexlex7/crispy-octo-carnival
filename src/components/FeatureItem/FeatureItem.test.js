import { render, screen } from '@testing-library/react';
import FeatureItem from './FeatureItem';
import { features } from '../../mock/features.json';

test('render feature item component', () => {
  const { asFragment } = render(<FeatureItem item={features[0]} />);
  const title = screen.getByText('Best Quality');
  expect(title).toBeInTheDocument();

  const text = screen.getByText(features[0].text);
  expect(text).toBeInTheDocument();

  const image = screen.getByRole('img', { description: features[0].title });
  expect(image).toBeInTheDocument();
  expect(asFragment()).toMatchSnapshot();
});
