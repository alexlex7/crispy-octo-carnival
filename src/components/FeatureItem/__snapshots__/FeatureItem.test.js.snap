// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render feature item component 1`] = `
<DocumentFragment>
  <li
    class="listItem"
  >
    <div
      class="thumb"
    >
      <img
        alt="Best Quality"
        sizes="(min-width: 1205px) 50px, (max-width: 1204px) 28px, 28px)"
        src="http://localhost:8080/images/features/mobileQuality.png"
        srcset="http://localhost:8080/images/features/mobileQuality.png 28w, http://localhost:8080/images/features/quality.png 50w"
        title="Best Quality"
      />
    </div>
    <div>
      <h6
        class="title"
      >
        Best Quality
      </h6>
      <p
        class="text"
      >
        All of our furniture uses the best materials and choices
      </p>
    </div>
  </li>
</DocumentFragment>
`;
